packages/dm-core-plugins/src/form/components/OpenObjectButton.tsx

```diff
@@ -2,17 +2,29 @@ import { Button } from '@equinor/eds-core-react'
 import React from 'react'
 import { useRegistryContext } from '../RegistryContext'
 
-export const OpenObjectButton = ({ namePath }: { namePath: string }) => {
+export const OpenObjectButton = ({
+  viewId,
+  namePath,
+  idReference,
+}: {
+  viewId: string
+  namePath: string
+  idReference?: string
+}) => {
   const { onOpen } = useRegistryContext()
 
   return (
     <Button
       variant="outlined"
       onClick={() =>
-        onOpen?.(namePath, {
-          type: 'ViewConfig',
-          scope: namePath,
-        })
+        onOpen?.(
+          viewId || namePath,
```

Comment:
are both viewId and namePath required props at the same time? Or should only one be passed in at a time?

If the second case is true, the correct type would be:

```TypeScript
type Props =
      ({
          viewId: string
          namePath?: never
        }
      | {
          viewId?: never;
          namePath: string;
        })
      & { idReference?: string };
```
This way, either viewId or namePath, but not both.

Reply:
Both are required